inc/utility/class-handlestransfers.php

```diff
@@ -0,0 +1,347 @@
+		] );
+
+		if ( ! empty( $upload['error'] ) ) {
+			$_SESSION['pb_errors'][] = $upload['error'];
+
+			return false;
+		}
+
+		return $upload;
+	}
+
+	/**
+	 * Returns the list of contributors to import
+	 *
+	 * @param array $upload
+	 * @return array
+	 */
+	public function getImportableItems( $upload ) {
+		$items = [];
+		$invalid_rows = [];
+
+		foreach ( json_decode( file_get_contents( $upload['file'] ) ) as $key => $item ) {
+			$item = (array) $item;
+
+			if ( ! isset( $item['name'], $item['slug'] ) ) {
+				$invalid_rows[] = $key + 1;
+
+				continue;
+			}
+
+			$items[] = $item;
+		}
+
+		if ( ! empty( $invalid_rows ) ) {
+			$message = __( 'One or more rows were not imported because they have missing information. Please, check rows %s', 'pressbooks' );
+
+			$_SESSION['pb_errors'][] = sprintf( $message, implode( ',', $invalid_rows ) );
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Imports the list of contributors.
+	 *
+	 * @param array $items
+	 */
+	public function handleImport( $items ) {
+		$importable_fields = $this->getTransferableFields();
+		$changed = false;
+
+		foreach ( $items as $item ) {
+			$term = get_term_by( 'slug', $item['slug'], self::TAXONOMY );
+
+			if ( ! $term ) {
+				$results = wp_insert_term( sanitize_text_field( $item['name'] ), self::TAXONOMY, [
+					'slug' => sanitize_text_field( $item['slug'] ),
+				] );
+			} else {
+				$results = wp_update_term( $term->term_id, self::TAXONOMY, [
+					'name' => sanitize_text_field( $item['name'] ),
+					'slug' => sanitize_text_field( $item['slug'] ),
+				] );
+			}
+
+			foreach ( $importable_fields as $field ) {
+				if ( ! isset( $item[ $field ] ) ) {
+					continue;
+				}
+
+				if ( empty( $item[ $field ] ) ) {
+					continue;
+				}
+
+				if ( in_array( $field, $this->getUrlFields(), true ) && ! \Pressbooks\Sanitize\validate_url_field( $item[ $field ] ) ) {
+					$changed = true;
+					$item[ $field ] = '';
+				}
+
+				if ( false !== strpos( $field, 'picture' ) ) {
+					// Skip the picture if we are unable to get the src url.
+					$src = $this->handleImage( $item[ $field ] );
+
+					if ( ! $src ) {
+						continue;
+					}
+
+					$item[ $field ] = $src;
+				}
+
+				$term
+					? update_term_meta( $results['term_id'], $field, sanitize_text_field( $item[ $field ] ) )
+					: add_term_meta( $results['term_id'], sanitize_text_field( $field ), $item[ $field ] );
+			}
+		}
+
+		if ( $changed ) {
+			$_SESSION['pb_notices'][] = __( 'Values for one or more of the imported contributors were altered by our validation routine. You may want to check the values for the newly imported contributors.', 'pressbooks' );
+		}
+	}
+
+	/**
+	 * Creates a new image based on the url provided during import.
+	 *
+	 * @param string $url
+	 * @return false|string
+	 */
+	public function handleImage( $url ) {
+		$parts = explode( '?', $url );
+		$parts = explode( '#', $parts[0] );
+		$parts = explode( '/', $parts[0] );
+
+		$filename = sanitize_file_name( end( $parts ) );
```

Comment:
Cara, I think we want to make sure it is a valid URL as well: https://github.com/pressbooks/pressbooks/blob/dev/inc/class-contributors.php#L320